ws/filetransfer/model.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package filetransfer
+
+const (
+	MessageTypeGetFile          = "get_file"
+	MessageTypePutFile          = "put_file"
+	MessageTypeStatFile         = "file_info_req"
+	MessageTypeStatFileResponse = "file_info_res"
+	MessageTypeFileChunk        = "file_chunk"
+	MessageTypeError            = "error"
+)
+
+// The Error struct is passed in the Body of MsgProto in case the message type is ErrorMessage
+type Error struct {
+	// The error description, as in "Permission denied while opening a file"
+	Error string `msgpack:"err" json:"error"`
+	// Type of message that raised the error
+	MessageType string `msgpack:"msgtype,omitempty" json:"message_type,omitempty"`
+	// Message id is passed in the MsgProto Properties, and in case it is available and
+	// error occurs it is passed for reference in the Body of the error message
+	MessageId string `msgpack:"msgid,omitempty" json:"message_id,omitempty"`
+}
+
+// Get file requests the flow of MessageTypeFileChunk messages to be started from the remote end
+type GetFile struct {
+	// The file path to the file we are requesting
+	Path string `msgpack:"filepath,omitempty" json:"file_path,omitempty"`
+}
+
+// Put file marks the flow of MessageTypeFileChunk messages to be started to the remote end
+type PutFile struct {
+	// The file path to the file we are requesting
+	Path string `msgpack:"filepath,omitempty" json:"file_path,omitempty"`
+}
+
+// Stat file requests the file stat structure from the remote end
+type StatFile struct {
+	// The file path to the file we are requesting
+	Path string `msgpack:"filepath,omitempty" json:"file_path,omitempty"`
+}
+
+// Stat file response is the reply to the StatFile from the remote end
+type StatFileResponse struct {
+	// The file path to the file we are sending status for
+	Path string `msgpack:"filepath,omitempty" json:"file_path,omitempty"`
+	// The file size
+	Size int64 `msgpack:"filesize,omitempty" json:"file_size,omitempty"`
+	// The file type
+	Type int `msgpack:"filetype,omitempty" json:"file_type,omitempty"`
+	// The file owner
+	Owner string `msgpack:"fileowner,omitempty" json:"file_owner,omitempty"`
+	// The file group
+	Group string `msgpack:"filegroup,omitempty" json:"file_group,omitempty"`
+}
```

Comment:
```suggestion
// FileInfo is the response to the StatFile request from the client (UI) side.
type FileInfo struct {
	// The file path to the file we are sending status for
	Path string `msgpack:"path,omitempty" json:"path,omitempty"`
	// The file size
	Size int64 `msgpack:"size,omitempty" json:"size,omitempty"`
	// The file type
	Type int `msgpack:"type,omitempty" json:"type,omitempty"`
	// The file owner
	Owner string `msgpack:"owner,omitempty" json:"owner,omitempty"`
	// The file group
	Group string `msgpack:"group,omitempty" json:"group,omitempty"`
}
```

Reply:
Also, should `Type` be a string property? Having it as an `int` may lead to compatibility issues if we plan on making it cross OS compatible in the future.

Reply:
no problema.